src/pages/ServiceGraph/ServiceGraphPage.tsx

```diff
@@ -69,6 +69,13 @@ export default class ServiceGraphPage extends React.Component<RouteComponentProp
     this.setState({ summaryData: data });
   };
 
+  shouldComponentUpdate(nextProps: any, nextState: any) {
+    if (this.state.summaryData.summaryType !== nextState.summaryData.summaryType) {
```

Comment:
I think that it might be better to return:
```typescript
return this.state.isLoading || nextState.isLoading;
```

We only want to update the component when we are leaving an `isLoading` or entering into a `isLoading`.
Briefly tested, but looks OK.

Maybe further refining to something like
```typescript
return this.state.isLoading !== nextState.isLoading;
```
"When leaving or entering into an isLoading state"

Reply:
Sorry, I used this into `CytoscapeLayout.tsx`